### DISCUSSION TITLE
Is there a way to map `blog/1.example.md` to `/blog/example`?

### DISCUSSION THREAD

**Discussion by aiktb** [vuejs/vitepress#3198] [Q&A] [answered]
[Nuxt Content](https://content.nuxt.com/usage/content-directory) has a feature like this, can it be implemented in VitePress?

![image](https://github.com/vuejs/vitepress/assets/104723527/942c5383-e7c4-490c-8b03-06f918385b52)

**Comment by Octobug** [comment, marked as answer]
Yes, there is.

```js
// .vitepress/config.js
export default defineConfig({
  title: "My Awesome Project",
  ...
  rewrites: {
    ':blog/:num.:title.md': ':blog/:title.md', // just ignore ":num." which matches `1.`
  }
})
```

<img width="1339" alt="image" src="https://github.com/vuejs/vitepress/assets/8007022/2d8c4d83-95ac-4a58-8c88-bba68e89e737">


Visit <https://vitepress.dev/guide/routing#route-rewrites> for more details.

**Reply by aiktb** [reply]
Thank you very much!

**Comment by aiktb** [comment]
> Yes, there is.
> 
> ```js
> // .vitepress/config.js
> export default defineConfig({
>   title: "My Awesome Project",
>   ...
>   rewrites: {
>     ':blog/:num.:title.md': ':blog/:title.md', // just ignore ":num." which matches `1.`
>   }
> })
> ```
> 
> <img alt="image" width="1339" src="https://user-images.githubusercontent.com/8007022/281610711-2d8c4d83-95ac-4a58-8c88-bba68e89e737.png">
> Visit https://vitepress.dev/guide/routing#route-rewrites for more details.
Hello, I have a question, I have an index page built with `createContentLoader`, the URL taken from it is in the format `/blog/1.example`, in dev mode I can observe that when clicking on this link the page undergoes a redirection targeting `/blog/example`, which works fine.
In production, no redirection is triggered, resulting in a 404 error.

I could simply modify this URL to match the correct format to eliminate the 404, but I'm wondering if this is a bug?

**Reply by Octobug** [reply]
I've reproduced this issue. In my opinion, it should be regarded as a bug because it produces different outcomes between development and production.

**Reply by aiktb** [reply]
You are right, I will open an issue for this.
Please remind me if you want to do the same.

**Reply by Octobug** [reply]
I'll keep up with the issue you are going to open ;)

**Reply by aiktb** [reply]
There is an open issue: https://github.com/vuejs/vitepress/issues/3062

**Comment by aiktb** [comment]
According to the documentation, this does not appear to be a bug.
https://vitepress.dev/guide/data-loading#createcontentloader
```ts
interface ContentData {
  // mapped URL for the page. e.g. /posts/hello.html (does not include base)
  // manually iterate or use custom `transform` to normalize the paths
  url: string
}
```
